packages/flutter_tools/lib/src/build_system/targets/web.dart

```diff
@@ -328,18 +329,32 @@ class Dart2WasmTarget extends Dart2WebTarget {
 /// Unpacks the dart2js or dart2wasm compilation and resources to a given
 /// output directory.
 class WebReleaseBundle extends Target {
-  WebReleaseBundle(List<WebCompilerConfig> configs) : this._withTargets(
-    configs.map((WebCompilerConfig config) =>
+  WebReleaseBundle(List<WebCompilerConfig> configs) : this._(
+    compileTargets: configs.map((WebCompilerConfig config) =>
       switch (config) {
         WasmCompilerConfig() => Dart2WasmTarget(config),
         JsCompilerConfig() => Dart2JSTarget(config),
       }
-    ).toList()
+    ).toList(),
   );
 
-  const WebReleaseBundle._withTargets(this.compileTargets);
+  WebReleaseBundle._({
+    required this.compileTargets,
+  }) : templatedFilesTarget = WebTemplatedFiles(generateBuildConfigString(compileTargets));
+
+  static String generateBuildConfigString(List<Dart2WebTarget> compileTargets) {
+    final List<Map<String, Object?>> buildDescriptions = compileTargets.map(
+      (Dart2WebTarget target) => target.buildConfig
+    ).toList();
+    final Map<String, Object?> buildConfig = <String, Object?>{
+      'engineRevision': globals.flutterVersion.engineRevision,
+      'builds': buildDescriptions,
+    };
+    return 'if (!_flutter) _flutter = {}; _flutter.buildConfig = ${jsonEncode(buildConfig)};';
```

Comment:
```suggestion
    return '''
if (!window._flutter) {
  window._flutter = {};
}
_flutter.buildConfig = ${jsonEncode(buildConfig)};
''';
```

(Not sure I'm doing multi-line strings right :P)